config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_dataformrepositories.dataform.cnrm.cloud.google.com.yaml

```diff
@@ -59,29 +59,134 @@ spec:
             type: object
           spec:
             properties:
+              displayName:
+                description: Optional. The repository's user-friendly name.
+                type: string
               gitRemoteSettings:
                 description: Optional. If set, configures this repository to be linked
                   to a Git remote.
                 properties:
-                  authenticationTokenSecretVersion:
+                  authenticationTokenSecretVersionRef:
                     description: The name of the Secret Manager secret version to
                       use as an authentication token for Git operations. Must be in
                       the format projects/* /secrets/* /versions/*.
-                    type: string
+                    oneOf:
+                    - not:
+                        required:
+                        - external
+                      required:
+                      - name
+                    - not:
+                        anyOf:
+                        - required:
+                          - name
+                        - required:
+                          - namespace
+                      required:
+                      - external
+                    properties:
+                      external:
+                        description: ' If provided must be in the format `projects/*/secrets/*/versions/*`.'
+                        type: string
+                      name:
+                        description: The `name` field of a `SecretManagerSecretVersion`
+                          resource.
+                        type: string
+                      namespace:
+                        description: The `metadata.namespace` field of a `SecretManagerSecretVersion`
+                          resource.
+                        type: string
+                    type: object
                   defaultBranch:
                     description: The Git remote's default branch name.
                     type: string
-                  tokenStatus:
-                    description: Indicates the status of the Git access token. https://cloud.google.com/dataform/reference/rest/v1beta1/projects.locations.repositories#TokenStatus.
-                    type: string
+                  sshAuthenticationConfig:
+                    description: Authentication fields for remote uris using SSH protocol.
+                    properties:
+                      hostPublicKey:
+                        description: Content of a public SSH key to verify an identity
+                          of a remote Git host.
+                        type: string
+                      userPrivateKeySecretVersionRef:
+                        description: The name of the Secret Manager secret version
+                          to use as a ssh private key for Git operations. Must be
+                          in the format projects/*/secrets/*/versions/* .
```

Comment:
Nit: we should probably remove the "Must be in the format" here from the docstring, not a blocker for alpha though  (and we might want a API linter in future!)

Reply:
As in we shouldn't share the expected format in the doc?

Reply:
We should, but it goes on `external`, not on the parent ref